services/storage/src/simcore_service_storage/modules/celery/_task.py

```diff
@@ -77,41 +61,92 @@ def wrapper(task: AbortableTask, *args: P.args, **kwargs: P.kwargs) -> R:
     return decorator
 
 
+def _error_handling(max_retries: NonNegativeInt, delay_between_retries: timedelta):
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
+        @wraps(func)
+        def wrapper(task: Task, *args: Any, **kwargs: Any) -> Any:
+            try:
+                return func(task, *args, **kwargs)
+            except Exception as exc:
+                exc_type = type(exc).__name__
+                exc_message = f"{exc}"
+                _logger.exception(
+                    "Task %s failed with exception: %s:%s",
+                    task.request.id,
+                    exc_type,
+                    exc_message,
+                )
+
+                # NOTE: since celery does a wonderful job when serializing exceptions
+                # by running it's magic, it looses the context of some errors
+                # this allows to recreate the same error in the caller side excatly as
+                # it was raised in this context
+                raise task.retry(
+                    max_retries=max_retries,
+                    countdown=delay_between_retries.total_seconds(),
+                    exc=_wrap_exception(exc),
+                )
```

Comment:
Please rewrite the NOTE. When I read it I do not understand what it is for.
so instead of Ignoring totally the task (which would stop the retries I guess?) you now explicitely ask for a retry with an interval. what is exc for?